### PR TITLE
#168 alter Expecting enum to be a bitflag

### PR DESCRIPTION
**Pull request by yuchunzhou**

### CODE REVIEW

File: ogma/src/lang/syntax/parse.rs [hunk 1]
```diff
@@ -105,25 +107,26 @@ pub fn definition_type<S: Into<Arc<str>>>(
...
-    /// Expecting a command/implementation.
-    Impl,
-    /// Expecting a type.
-    Type,
-    /// Expecting a term (the `bar` in `foo bar`).
-    ///
-    /// > Because commands can be inlined, if `Term` is encountered, impls would suit as well.
-    Term,
-    /// Expecting a Special Literal.
+bitflags! {
```
Comment (kurtlawrence) [inline ogma/src/lang/syntax/parse.rs:110]: Change to `bitflags::bitflags!`

File: ogma/src/lang/syntax/parse.rs [hunk 2]
```diff
@@ -1,8 +1,10 @@
 //! Parsing source into a AST.
+#![allow(non_upper_case_globals)]
 
 use crate::prelude::{ast::*, err, Definitions, HashSet};
 use ::kserd::Number;
 use ::libs::divvy::Str;
+use bitflags::bitflags;
```
Comment (kurtlawrence) [inline ogma/src/lang/syntax/parse.rs:7]: Removing import, it is only used once, so we can use qualified name for it.

File: ogma/src/lang/syntax/parse.rs [hunk 3]
```diff
@@ -1,8 +1,10 @@
 //! Parsing source into a AST.
+#![allow(non_upper_case_globals)]
```
Comment (kurtlawrence) [inline ogma/src/lang/syntax/parse.rs:2]: Remove

File: ogma/src/lang/syntax/parse.rs [hunk 4]
```diff
@@ -105,25 +107,26 @@ pub fn definition_type<S: Into<Arc<str>>>(
...
+    /// An indication of expected AST node type if parsing fails.
     ///
-    /// These are `#t`, `#f`, and `#i`.
-    SpecLiteral,
-    /// Unable to provide an AST expectation.
-    None,
+    /// This is a best guess based on parsing context.
+    pub struct Expecting: u32 {
+	/// Expecting a command/implementation.
+	const Impl = 0b00001;
```
Comment (kurtlawrence) [inline ogma/src/lang/syntax/parse.rs:116]: I am happy to capitalise the constants.

File: ogma/src/lang/syntax/parse.rs [hunk 5]
```diff
@@ -105,25 +107,26 @@ pub fn definition_type<S: Into<Arc<str>>>(
...
+	/// Expecting a term (the `bar` in `foo bar`).
+	///
+	/// > Because commands can be inlined, if `Term` is encountered, impls would suit as well.
+	const Term = 0b00100;
+	/// Expecting a Special Literal.
+	///
+	/// These are `#t`, `#f`, and `#i`.
+	const SpecLiteral = 0b01000;
+	/// Unable to provide an AST expectation.
+	const None = 0b10000;
```
Comment (kurtlawrence) [inline ogma/src/lang/syntax/parse.rs:128]: Please set `const NONE = 0;`
It will match **all** numbers, but also acts as the empty value.

**Review by kurtlawrence — CHANGES_REQUESTED** [review]
Thanks @yuchunzhou, I've just got a couple of changes that I would like implemented.

**Comment by yuchunzhou** [comment]
hi @kurtlawrence, the code is updated.

**Review by kurtlawrence — APPROVED** [review]

**Comment by kurtlawrence** [comment]
Thanks @yuchunzhou, I've merged it in, despite the CI failures. I've got a fix anyway coming through.

**Pull request merged**